### DISCUSSION TITLE
RFC: Always treat the argument of `to_timestamp(<number>)` as microsecs

### DISCUSSION THREAD

**Discussion by andylokandy** [databendlabs/databend#8160] [General]
`to_timestamp` in Snowflake and Databend treats its argument as seconds, milliseconds, or micros depending on the number scale ([detailed spec](https://docs.snowflake.com/en/sql-reference/functions/to_timestamp.html#:~:text=If%20the%20integer%20is,is%20treated%20as%20nanoseconds.)). Because the user has to control the behavior with the input value, it may not be error-prone enough, especially when converting a column of values. Therefore, I propose to fix the argument to microsecs.

**Comment by andylokandy** [comment]
cc @BohuTANG @sundy-li @Xuanwo

**Reply by sundy-li** [reply]
Agree. 

Related to issue: https://github.com/datafuselabs/databend/issues/5698  cc @wubx

**Reply by sundy-li** [reply]
But is there a better way to create a timestamp from number without place too much trailing zeros?

`to_timestamp(1665556629000000)`  

1665556629000000 is tooooo long to use.

**Reply by sundy-li** [reply]
Oh. I found the parser supported it.

```

MySQL [(none)]> select to_timestamp(1665556629e6);
+--------------------------------+
| to_timestamp(1665556629000000) |
+--------------------------------+
| 2022-10-12 06:37:09.000000     |
+--------------------------------+
1 row in set (0.024 sec)
```

**Reply by andylokandy** [reply]
in snowflake, to_timestamp could accept an extra argument `scale` for that purpose
